Code/Framework/AzFramework/Platform/Linux/AzFramework/Components/NativeUISystemComponent_Linux.cpp

```diff
@@ -0,0 +1,151 @@
+        {
+#if PAL_TRAIT_LINUX_WINDOW_MANAGER_XCB
+            return aznew XcbInputDeviceKeyboard(inputDevice);
+#elif PAL_TRAIT_LINUX_WINDOW_MANAGER_WAYLAND
+            #error "Linux Window Manager Wayland not supported."
+            return nullptr;
+#else
+            #error "Linux Window Manager not recognized."
+            return nullptr;
+#endif // PAL_TRAIT_LINUX_WINDOW_MANAGER_XCB
+        }
+    };
+
+    struct LinuxDeviceMouseImplFactory
+        : public InputDeviceMouse::ImplementationFactory
+    {
+        InputDeviceMouse::Implementation* Create(InputDeviceMouse& inputDevice) override
+        {
+#if PAL_TRAIT_LINUX_WINDOW_MANAGER_XCB
+            return XcbInputDeviceMouse::Create(inputDevice);
+#elif PAL_TRAIT_LINUX_WINDOW_MANAGER_WAYLAND
+#error "Linux Window Manager Wayland not supported."
+            return nullptr;
+#else
+#error "Linux Window Manager not recognized."
+            return nullptr;
+#endif // PAL_TRAIT_LINUX_WINDOW_MANAGER_XCB
+        }
+    };
+
+    struct LinuxNativeWindowFactory 
+        : public NativeWindow::ImplementationFactory
+    {
+        NativeWindow::Implementation* Create() override
+        {
+#if PAL_TRAIT_LINUX_WINDOW_MANAGER_XCB
+            return aznew XcbNativeWindow();
+#elif PAL_TRAIT_LINUX_WINDOW_MANAGER_WAYLAND
+            #error "Linux Window Manager Wayland not supported."
+            return nullptr;
+#else
+            #error "Linux Window Manager not recognized."
+            return nullptr;
+#endif // PAL_TRAIT_LINUX_WINDOW_MANAGER_XCB
+        }
+    };
+
+    Application::ImplementationFactory* NativeUISystemComponent::CreateApplicationImplementationFactory() const
+    {
+        return aznew LinuxApplicationImplFactory();
+    }
+
+    InputDeviceKeyboard::ImplementationFactory* NativeUISystemComponent::GetDeviceKeyboardImplementationFactory() const
+    {
+        return aznew LinuxDeviceKeyboardImplFactory();
+    }
+
+    InputDeviceMouse::ImplementationFactory* NativeUISystemComponent::GetDeviceMouseImplentationFactory() const
+    {
+        return aznew LinuxDeviceMouseImplFactory();
+    }
+
+    InputDeviceGamepad::ImplementationFactory* NativeUISystemComponent::GetDeviceGamepadImplentationFactory() const
+    {
+        return nullptr;
+    }
+
+    InputDeviceMotion::ImplementationFactory* NativeUISystemComponent::GetDeviceMotionImplentationFactory() const
+    {
+        return nullptr;
+    }
+
+    InputDeviceTouch::ImplementationFactory* NativeUISystemComponent::GetDeviceTouchImplentationFactory() const
+    {
+        return nullptr;
+    }
+    InputDeviceVirtualKeyboard::ImplementationFactory* NativeUISystemComponent::GetDeviceVirtualKeyboardImplentationFactory() const
+    {
+        return nullptr;
+    }
+    NativeWindow::ImplementationFactory* NativeUISystemComponent::GetNativeWindowImplementationFactory() const
+    {
+        return aznew LinuxNativeWindowFactory();
+    }
```

Comment:
These functions should all return unique pointers since they are ownership over to the caller.